genie-common-internal/src/main/java/com/netflix/genie/common/internal/services/impl/FileSystemJobArchiverImpl.java

```diff
@@ -83,7 +83,20 @@ public boolean archiveDirectory(final Path directory, final URI target) throws J
         private final Path source;
         private final Path target;
 
-        JobDirectoryCopier(final Path source, final Path target) {
+        JobDirectoryCopier(final Path source, final Path target) throws IOException {
+            // If the source doesn't exist or isn't a directory throw an exception
```

Comment:
I don't know that i agree with checking these here, since by the time you get to the actual copy they may no longer be true, so why bother.

Reply:
I actually added these checks after some integration tests failed due to some of these conditions